cpp/include/cudf/groupby.hpp

```diff
@@ -231,6 +231,11 @@ class groupby {
     std::vector<aggregation_request> const& requests,
     rmm::cuda_stream_view stream,
     rmm::mr::device_memory_resource* mr);
+
+  std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> sort_scan(
```

Comment:
A couple questions:
- Should we use `sort` in naming here? To implement `scan`, sorting is kind of necessary, but is it an explicit guarantee?
- Maybe rename `aggregation_{request, result}` to `groupby_op_{request, result}` as it is shared across `aggregation` and `scan`?